### DISCUSSION TITLE
STM32 b_u585i_iot02a_ns, problems with gpio_pin_interrupt_configure

### DISCUSSION THREAD

**Discussion by JiiZR** [zephyrproject-rtos/zephyr#58141] [Q&A]
**Describe the bug**
I was trying to use two GPIO interrupts, different port but same pin number. Second pin gpio_pin_interrupt_configure
fails: failed to configure interrupt gpio_e, ret:-16

Interrupts are working, tested separately but if port number is same then second always gives -16, EBUSY.
 Is this a bug or platform limitation or user error ?

**To Reproduce**
1. west build -p auto -b b_u585i_iot02a_ns samples\basic\blinky
2. west flash
3. See error

**Expected behavior**
No error prints and both interrupts should work

**Impact**
Limits pin usage

**Logs and console output**

*** Booting Zephyr OS build zephyr-v3.3.0-4047-g49389b546a4a ***
gpio_d intr init done
failed to configure interrupt gpio_e, ret:-16
gpio_e intr init done

**Environment (please complete the following information):**
 - OS: Windows
 - Toolchain Zephyr SDK,
 - latest codes

**Additional context**
Add any other context that could be relevant to your issue, such as pin setting,
target configuration, ...
[main.c.txt](https://github.com/zephyrproject-rtos/zephyr/files/11514733/main.c.txt)

**Comment by erwango** [comment]
Proposed code is doing the following:
```
	ret = gpio_pin_interrupt_configure(gpio_d_dev, 15, GPIO_INT_EDGE_FALLING );
        ...
      	ret = gpio_pin_interrupt_configure(gpio_e_dev, 15, GPIO_INT_EDGE_FALLING);
```

Please note (see ref man) that there is only one interrupt line  for all GPIO ports.
Trying to use line 14 on gpio_e will work.
See https://github.com/zephyrproject-rtos/zephyr/issues/45234 for a, finally impossible, attempt to work around it.

Note, I'm converting this to discussion.

**Reply by JiiZR** [reply]
Hi

Thanks for the information.

After reading https://github.com/zephyrproject-rtos/zephyr/issues/45234
I would like to be sure that
this is correct interpretation:

It is not possible to use same pin numbers from from different port as separate interrupts ?

**Reply by erwango** [reply]
> It is not possible to use same pin numbers from from different port as separate interrupts ?

This is correct.

**Reply by JiiZR** [reply]
Ok, thanks.

**Reply by zafersn** [reply]
hi @erwango  @JiiZR , I am a little bit confused, as far as I understand here 
https://github.com/zephyrproject-rtos/zephyr/issues/45234#issuecomment-1171254614 
what you proposed, it is possible to share the interrupt line for the same pin numbers in different ports. Still, currently, I cant see this configuration in the config `CONFIG_EXTI_STM32_ONE_IRQ=y` I just need to have an interrupt doesn't matter which pin generates, and then I can poll it.  So isn't this possible?

if so, What should be the initialising configs?

lets say status1_pin =PB14, status2_pin PD14
```
if (!gpio_is_ready_dt(&cfg->status1_pin)) 
	{
		if (cfg->status1_pin.port) {
			LOG_ERR("%s: device %s is not ready", dev->name,
						cfg->status1_pin.port->name);
			ret = -ENODEV;
			goto error;
		}
		ret = -ENODEV;
		LOG_DBG("%s: status1_pin not defined in DT", dev->name);
		goto error;
	}
	if (!gpio_is_ready_dt(&cfg->status2_pin)) 
	{
		if (cfg->status2_pin.port) {
			LOG_ERR("%s: device %s is not ready", dev->name,
						cfg->status2_pin.port->name);
			ret = -ENODEV;
			goto error;
		}
		ret = -ENODEV;
		LOG_DBG("%s: status2_pin not defined in DT", dev->name);
		goto error;
	}
           ret = gpio_pin_configure_dt(&cfg->status1_pin, GPIO_INPUT);
		if (ret < 0) {
			LOG_ERR("Could not configure gpio 1");
			goto error;
		}
		ret = gpio_pin_configure_dt(&cfg->status2_pin, GPIO_INPUT);
		if (ret < 0) {
			LOG_ERR("Could not configure gpio 2");
			goto error;
		}

               gpio_init_callback(&aem10941->gpio_cb,
				   aem10941_gpio_callback,
				   BIT(cfg->status1_pin.pin) | BIT(cfg->status2_pin.pin));

             ret = gpio_add_callback(cfg->status1_pin.port, &aem10941->gpio_cb);
		if (ret < 0) {
			LOG_ERR("Could not set gpio callback 1");
			goto error;
		}
		ret = gpio_add_callback(cfg->status2_pin.port, &aem10941->gpio_cb);
		if (ret < 0) {
			LOG_ERR("Could not set gpio callback 2");
			goto error;
		}

               ret = gpio_pin_interrupt_configure_dt(&cfg->status1_pin,
							GPIO_INT_EDGE_TO_ACTIVE);
		if (ret < 0) {
				LOG_ERR("Could not set gpio int 1");
				goto error;
		}
		ret = gpio_pin_interrupt_configure_dt(&cfg->status2_pin,
							GPIO_INT_EDGE_TO_ACTIVE);
		if (ret < 0) {
				LOG_ERR("Could not set gpio int 2 %d", ret);
				goto error;
		}
```

This doesn't work `ret = gpio_pin_interrupt_configure_dt(&cfg->status2_pin, GPIO_INT_EDGE_TO_ACTIVE);` return -EBUSY
. If I remove intrpt config for status2 then, intrpt doesn't work

**Reply by erwango** [reply]
@zafersn By design on STM32, there is only one IRQ line per pin number (could also be same line for several pins) used across all GPIO ports. 

For Instance IRQ line 4 used for pins 4. Applying to all GPIO ports, hence PA4 to PH4.
(It could also be IRQ line 11 used for pins 11 to 15. Applying to all GPIO ports, hence PA11>PA15 to PH11>15).

Today, there is no SW way to understand on which port an IRQ raise on a given line comes from.

CONFIG_EXTI_STM32_ONE_IRQ was an attempt to make it working, but it wasn't merged: https://github.com/zephyrproject-rtos/zephyr/pull/47108.
